lib/k8s/typed_resource.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module K8s
+  # generic typed resource
+  class TypedResource < Dry::Struct
+    using K8s::Util::HashDeepMerge
+
+    transform_keys do |key|
+      key = key.sub('$', '_') if key.is_a?(String)
```

Comment:
Ruby methods can't start with a number and methods that start with Capital can't be invoked without `()`. Containing a dot or @ is also bad, also `[ ] ( ) + - *` will be problematic.

`.sub` will only replace the first occurence and I believe the problem is the same when the key is a symbol. 

Maybe the generator should skip creating models for objects that have attributes that can't be used as method names, won't it make the model unusable because it can't restore the original key when sending data to kube?

Or maybe the generator needs some way to generate custom `transform_keys` blocks for models with funny keys and rig the `.to_h` to replace the keys with the original ones for sending?

Reply:
`$` was the only thing that did not pass.. we can improve this if we see errors.